mk/spksrc.wheel-download.mk

```diff
@@ -0,0 +1,92 @@
+### Wheel rules
+# Download wheels for modules listed in WHEELS. 
+#
+# Targets are executed in the following order:
+#  wheel_download_msg_target
+#  pre_wheel_download_target   (override with PRE_WHEEL_DOWNLOAD_TARGET)
+#  wheel_download_target       (override with WHEEL_DOWNLOAD_TARGET)
+#  post_wheel_download_target  (override with POST_WHEEL_DOWNLOAD_TARGET)
+# Variables:
+#  REQUIREMENT             Requirement formatted wheel information
+#  WHEEL_NAME              Name of wheel to process
+#  WHEEL_VERSION           Version of wheel to process (can be empty)
+#  WHEEL_TYPE              Type of wheel to process (abi3, crossenv, pure)
+
+ifeq ($(WHEEL_VERSION),)
+WHEEL_DOWNLOAD_COOKIE = $(WORK_DIR)/.$(COOKIE_PREFIX)wheel_download-$(WHEEL_NAME)_done
+else
+WHEEL_DOWNLOAD_COOKIE = $(WORK_DIR)/.$(COOKIE_PREFIX)wheel_download-$(WHEEL_NAME)-$(WHEEL_VERSION)_done
+endif
+
+##
+
+ifeq ($(strip $(PRE_WHEEL_DOWNLOAD_TARGET)),)
+PRE_WHEEL_DOWNLOAD_TARGET = pre_wheel_download_target
+else
+$(PRE_WHEEL_DOWNLOAD_TARGET): wheel_download_msg_target
+endif
+ifeq ($(strip $(WHEEL_DOWNLOAD_TARGET)),)
+WHEEL_DOWNLOAD_TARGET = wheel_download_target
+else
+$(WHEEL_DOWNLOAD_TARGET): $(BUILD_WHEEL_DOWNLOAD_TARGET)
+endif
+ifeq ($(strip $(POST_WHEEL_DOWNLOAD_TARGET)),)
+POST_WHEEL_DOWNLOAD_TARGET = post_wheel_download_target
+else
+$(POST_WHEEL_DOWNLOAD_TARGET): $(WHEEL_DOWNLOAD_TARGET)
+endif
+
+wheel_download_msg_target:
+	@$(MSG) "Processing wheels of $(NAME)"
+
+pre_wheel_download_target: wheel_download_msg_target
+
+wheel_download_target: SHELL:=/bin/bash
+wheel_download_target:
+ifeq ($(wildcard $(PIP_DISTRIB_DIR)),)
+	@$(MSG) Creating pip download directory: $(PIP_DISTRIB_DIR)
+	@mkdir -p $(PIP_DISTRIB_DIR)
+endif
+ifeq ($(wildcard $(PIP_CACHE_DIR)),)
+	@$(MSG) Creating pip caching directory: $(PIP_CACHE_DIR)
+	@mkdir -p $(PIP_CACHE_DIR)
+endif
+	@$(MSG) Downloading wheel [$(WHEEL_NAME)], version [$(WHEEL_VERSION)] ; \
+#	$(MSG) requirement: [$(REQUIREMENT)] ; \
+#	$(MSG) requirement-grep-egg: [$$(grep -s egg <<< $(REQUIREMENT))] ; \
+#	$(MSG) name: [$(WHEEL_NAME)] ; \
+#	$(MSG) type: [$(WHEEL_TYPE)] ; \
+#	$(MSG) version: [$(WHEEL_VERSION)] ; \
+#	$(MSG) type: [$(WHEEL_TYPE)] ; \
```

Comment:
If this does not work, it might all be commented out (`\` wraps all on a single line, and all following the first # is comment)

_just a guess_

Reply:
yeah, well, i intend to remove that prior to merging.  These are left over debug traces.

Reply:
now removed